org.eclipse.xtext.xbase/generator/org/eclipse/xtext/xbase/GenerateXbase.java

```diff
@@ -39,7 +39,7 @@ public static void main(String[] args) {
 		final String root = "..";
 		final String projectName = "org.eclipse.xtext.xbase";
 		final String runtimeProject = root + "/" + projectName;
-		final String uiProject = root + "/../xtext-eclipse/" + projectName + ".ui";	
+		final String uiProject = System.getProperty("uiProject", root + "/../xtext-eclipse/" + projectName + ".ui");	
```

Comment:
hmmm i wonder why this is needed. shouldnt root be the parameter as it is for other workflows?

Reply:
runtimeProject and uiProject are in different repositories and there is no reliable way to navigate in the file system of the developers machine from one repo to another. Thus I need to pass the other projects path as an argument